src/storage/store/node_group.cpp

```diff
@@ -130,19 +133,48 @@ void NodeGroup::initializeScanState(Transaction*, const UniqLock& lock,
     TableScanState& state) const {
     auto& nodeGroupScanState = *state.nodeGroupScanState;
     nodeGroupScanState.chunkedGroupIdx = 0;
-    nodeGroupScanState.numScannedRows = 0;
+    nodeGroupScanState.nextRowToScan = 0;
     ChunkedNodeGroup* firstChunkedGroup = chunkedGroups.getFirstGroup(lock);
     initializeScanStateForChunkedGroup(state, firstChunkedGroup);
 }
 
+void applySemiMaskFilter(const TableScanState& state, row_idx_t numRowsToScan,
```

Comment:
How is this function related to GDS vertex property scanning?

Reply:
It's not, but the vertex property scanning needed a new overload of `NodeGroup::scan`, and I thought that new function should probably also support the semi masks and moved that code into a function so that it doesn't need to be duplicated.